src/Resolvers/Dig.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace RemotelyLiving\PHPDNS\Resolvers;
+
+use RemotelyLiving\PHPDNS\Entities\DNSRecordCollection;
+use RemotelyLiving\PHPDNS\Entities\DNSRecordType;
+use RemotelyLiving\PHPDNS\Entities\Hostname;
+use RemotelyLiving\PHPDNS\Factories\SpatieDNS;
+use RemotelyLiving\PHPDNS\Mappers\Dig as DigMapper;
+use RemotelyLiving\PHPDNS\Resolvers\Exceptions\QueryFailure;
+
+class Dig extends ResolverAbstract
+{
+    private const SUPPORTED_QUERY_TYPES = [
+        DNSRecordType::TYPE_A,
+        DNSRecordType::TYPE_AAAA,
+        DNSRecordType::TYPE_CNAME,
+        DNSRecordType::TYPE_NS,
+        DNSRecordType::TYPE_SOA,
+        DNSRecordType::TYPE_MX,
+        DNSRecordType::TYPE_SRV,
+        DNSRecordType::TYPE_TXT,
+        DNSRecordType::TYPE_CAA,
+        DNSRecordType::TYPE_NAPTR,
+    ];
+
+    /**
+     * @var \RemotelyLiving\PHPDNS\Factories\SpatieDNS
+     */
+    private $spatieDNSFactory;
+
+    /**
+     * @var \RemotelyLiving\PHPDNS\Mappers\Dig
+     */
+    private $mapper;
+
+    /**
+     * @var \RemotelyLiving\PHPDNS\Entities\Hostname|null
+     */
+    private $nameserver;
+
+    public function __construct(
+        SpatieDNS $spatieDNSFactory = null,
+        DigMapper $mapper = null,
+        Hostname $nameserver = null
```

Comment:
here is where you could specify an alternative nameserver...come to think of it, it should be the first constructor argument as it's the only one you really care about constructing